client/client.go

```diff
@@ -594,3 +716,85 @@ func stripQuotes(word string) string {
 	}
 	return word
 }
+
+type ThreadSafeCache struct {
```

Comment:
why add this too client instead of as separate cache?
do we have a non thread safe cache? should we just name it cache and allow  have thread-safe implementation for the methods/functions?